featuretests/cmd_juju_upgrade_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package featuretests
+
+import (
+	"fmt"
+
+	"github.com/juju/cmd"
+	"github.com/juju/cmd/cmdtesting"
+	"github.com/juju/loggo"
+	jc "github.com/juju/testing/checkers"
+	"github.com/juju/version"
+	gc "gopkg.in/check.v1"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/cmd/juju/commands"
+	jujutesting "github.com/juju/juju/juju/testing"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/testing"
+	"github.com/juju/juju/testing/factory"
+	coreversion "github.com/juju/juju/version"
+)
+
+type cmdUpgradeSuite struct {
+	jujutesting.JujuConnSuite
+
+	hostedModelUser    string
+	hostedModelUserTag names.UserTag
+
+	hostedModel string
+}
+
+func (s *cmdUpgradeSuite) SetUpTest(c *gc.C) {
+	v, _ := version.Parse(oldVersion)
+	s.PatchValue(&coreversion.Current, v)
+
+	s.JujuConnSuite.SetUpTest(c)
+
+	s.AddToolsToState(c, version.MustParseBinary(fmt.Sprintf("%v-precise-amd64", newVersion)))
+	s.AddToolsToState(c, version.MustParseBinary(fmt.Sprintf("%v-trusty-amd64", newVersion)))
+	s.AddToolsToState(c, version.MustParseBinary(fmt.Sprintf("%v-xenial-amd64", newVersion)))
+	s.AddToolsToState(c, version.MustParseBinary(fmt.Sprintf("%v-yakkety-amd64", newVersion)))
```

Comment:
seems odd to need all of these, and yet still not cover zesty or artful.
Would it make more sense to just upload whatever we decide coreversion.Current is supposed to be?